src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -225,6 +226,10 @@ protected function refreshUser(TokenInterface $token)
                 $this->logger->debug('Token was deauthenticated after trying to refresh it.');
             }
 
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(new UserHasChangedEvent($token, $user, $userDeauthenticated, $userNotFoundByProvider), UserHasChangedEvent::class);
+            }
+
```

Comment:
I'm wondering if we could return the token in case a listener reauthenticated it?